### ISSUE TITLE
Style/FormatStringToken unannotated - false negative for Date/Time format

### ISSUE THREAD

**Issue by nattfodd** [bbatsov/rubocop#5223]
Rubocop 0.52.0 [introduced](https://github.com/bbatsov/rubocop/commit/b32b260df17e6370bff872ef4a2b551880f3411e) check for unannotated style, and now it throws error on every timestamp expression with default configuration. It makes sense for sting interpolation, but it does not for string to date conversion pattern.

## Expected behavior
... not to warn about `'%Y-%m-%dT%H:%M:%S.%L%z'`

## Actual behavior

It throws `Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s).`

## Steps to reproduce the problem
Code that causes warning:
```ruby
Time.strptime('2017-10-13T12:56:33.000Z', '%Y-%m-%dT%H:%M:%S.%L%z')
```

## RuboCop version
```
$ rubocop -V
0.52.0 (using Parser 2.4.0.2, running on ruby 2.4.2 x86_64-darwin17)
```

**Comment by jfelchner** [comment]
@nattfodd it's not just date strings, it's _any_ string that contains a percent sign.  Unless the string is being used via `%` or `printf` or something like that, the string should be ignored.

Example of a current false positive:

```ruby
class Foo
  def foo
    my_var = '%something%'
  end
end
```

**Comment by vinagrito** [comment]
Hey guys. I can see that it was merged into 0.52.1 
Locally I still get `Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over unannotated tokens (like %s)` when calling:
```
Time.strptime(time_date_joined, "%H:%M-%d/%m/%Y")
```
Am I missing something?

```
bundle show rubocop
=> .bundle/gems/rubocop-0.52.1
```

**Comment by AlexWayfer** [comment]
@vinagrito #5398

**Comment by vinagrito** [comment]
@AlexWayfer I knew I was missing something :). Thx!
